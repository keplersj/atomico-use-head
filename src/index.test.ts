import { jest } from "@jest/globals";
import { createHooks } from "atomico/test-hooks";

afterEach(() => {
  document.getElementsByTagName("html")[0].innerHTML = "";
});

describe("useHead Hook", () => {
  describe("module", () => {
    it("exports match expectations", async () => {
      const module = await import("./index");

      expect(module).toMatchInlineSnapshot(`
        Object {
          "useHead": [Function],
        }
      `);
    });
  });

  describe("hook, no initial state", () => {
    it("returns a result object when run", async () => {
      const host = document.createElement("div");
      host.attachShadow({ mode: "open" });
      document.body.appendChild(host);
      const hooks = createHooks(jest.fn(), host);

      const { useHead } = await import("./index");
      const head = hooks.load(() => useHead({}));

      expect(head).toMatchSnapshot();

      expect(document.documentElement).toMatchSnapshot();
    });
  });

  describe("hook, with initial state", () => {
    it("returns a result object when run", async () => {
      const host = document.createElement("div");
      host.attachShadow({ mode: "open" });
      document.body.appendChild(host);
      const hooks = createHooks(jest.fn(), host);

      const { useHead } = await import("./index");
      const head = hooks.load(() =>
        useHead({
          title: "Test",
          meta: [
            {
              name: "description",
              content: "Test Description",
            },
          ],
          link: [
            {
              rel: "apple-touch-icon",
              sizes: "48x48",
              href: "/icons/icon-48x48.png",
            },
          ],
          base: {
            href: "/",
            target: "_blank",
          },
          style: [
            {
              children: "body {color: red}",
            },
          ],
          script: [
            {
              as: "script",
              rel: "preload",
              href: "/framework.js",
            },
          ],
          htmlAttrs: {
            lang: "en",
          },
          bodyAttrs: {
            class: "root",
          },
        })
      );

      hooks.cleanEffects();

      expect(head).toMatchSnapshot();

      expect(document.documentElement).toMatchSnapshot();

      hooks.cleanEffects(true);

      expect(document.documentElement).toMatchSnapshot();
    });
  });
});
