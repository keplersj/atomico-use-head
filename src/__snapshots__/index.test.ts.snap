// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useHead Hook hook, no initial state returns a result object when run 1`] = `
Object {
  "state": Object {
    "base": undefined,
    "bodyAttrs": undefined,
    "htmlAttrs": undefined,
    "link": Array [],
    "meta": Array [],
    "script": Array [],
    "style": Array [],
    "title": undefined,
  },
}
`;

exports[`useHead Hook hook, no initial state returns a result object when run 2`] = `
<html>
  <head />
  <body>
    <div />
  </body>
</html>
`;

exports[`useHead Hook hook, with initial state returns a result object when run 1`] = `
Object {
  "state": Object {
    "base": Object {
      "href": "/",
      "target": "_blank",
    },
    "bodyAttrs": Object {
      "class": "root",
    },
    "htmlAttrs": Object {
      "lang": "en",
    },
    "link": Array [
      Object {
        "href": "/icons/icon-48x48.png",
        "rel": "apple-touch-icon",
        "sizes": "48x48",
      },
    ],
    "meta": Array [
      Object {
        "content": "Test Description",
        "name": "description",
      },
    ],
    "script": Array [
      Object {
        "as": "script",
        "href": "/framework.js",
        "rel": "preload",
      },
    ],
    "style": Array [
      Object {
        "children": "body {color: red}",
      },
    ],
    "title": "Test",
  },
}
`;

exports[`useHead Hook hook, with initial state returns a result object when run 2`] = `
<html
  lang="en"
>
  <head>
    Mark {}
    <title
      data-atomico-helmet="true"
    >
      Mark {}
      Test
      Mark {}
    </title>
    <meta
      content="Test Description"
      data-atomico-helmet="true"
      name="description"
    />
    <link
      data-atomico-helmet="true"
      href="/icons/icon-48x48.png"
      rel="apple-touch-icon"
      sizes="48x48"
    />
    <base
      data-atomico-helmet="true"
      href="/"
      target="_blank"
    />
    <style
      data-atomico-helmet="true"
    >
      Mark {}
      body {color: red}
      Mark {}
    </style>
    <script
      as="script"
      data-atomico-helmet="true"
      href="/framework.js"
      rel="preload"
    />
    Mark {}
  </head>
  <body
    class="root"
  >
    <div />
  </body>
</html>
`;

exports[`useHead Hook hook, with initial state returns a result object when run 3`] = `
<html
  lang=""
>
  <head>
    Mark {}
    Mark {}
  </head>
  <body
    class=""
  >
    <div />
  </body>
</html>
`;
